src/main/java/org/radarcns/management/service/SubjectService.java

```diff
@@ -152,18 +156,34 @@ public SubjectDTO updateSubject(SubjectDTO subjectDto) {
             source.setAssigned(true);
         }
         // update participant role
-        Set<Role> managedRoles = subject.getUser().getRoles().stream()
-                .filter(r -> !AuthoritiesConstants.PARTICIPANT.equals(r.getAuthority().getName()))
-                .collect(Collectors.toSet());
-        managedRoles.add(getProjectParticipantRole(subjectDto.getProject()));
+        Set<Role> managedRoles = updateParticipantRoles(subject, subjectDto);
```

Comment:
Minor thing: I was a bit confused until i realized `subject` refers to the 'old' subject and `subjectDto` refers to the updated info. Perhaps the variable names could be `oldSubject` and `newSubject` or something along those lines.